### PR TITLE
fix(graphql-model-transformer): override resource logical id to fix v1 to v2 transformer migration

### PR DESCRIPTION
**Pull request by lazpavel**
#### Description of changes
- sets resolver and data source logical IDs as V1 to allow proper migration
- adds migration test for @model
- fixes auth resolver lookup when queries are renamed

#### Description of how you validated changes
- `yarn test` passes
- manual testing
- e2e migration for @model with key timestamps and queries override from v1 to v2 passes

#### Checklist
- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-graphql-transformer-core/src/transform-host.ts [hunk 1]
```diff
@@ -131,6 +131,7 @@ export class DefaultTransformHost implements TransformHostProvider {
   public addResolver(
     typeName: string,
     fieldName: string,
+    resolverLogicalId: string,
```
Comment (edwardfoyle) [inline packages/amplify-graphql-transformer-core/src/transform-host.ts:134]: Should this be an option param or is it required in all cases?

File: packages/amplify-graphql-model-transformer/src/graphql-model-transformer.ts [hunk 2]
```diff
@@ -347,6 +366,7 @@ export class ModelTransformer extends TransformerModelBase implements Transforme
     type: ObjectTypeDefinitionNode,
     typeName: string,
     fieldName: string,
+    resolverLogicalId: string,
```
Comment (edwardfoyle) [inline packages/amplify-graphql-model-transformer/src/graphql-model-transformer.ts:369]: could we compute the logical ID of the resolver inside these functions using the typeName and fieldName rather than passing it in? Seems like that would significantly reduce the number of lines of code that need to change

Comment (lazpavel) [reply packages/amplify-graphql-model-transformer/src/graphql-model-transformer.ts:369]: typeName in the addResolver is either Query/Mutation/Sync, not the actual object type

File: packages/amplify-graphql-transformer-core/src/transform-host.ts (hunk 1 above)
Comment (lazpavel) [reply packages/amplify-graphql-transformer-core/src/transform-host.ts:134]: to me it looks that is should be required in all cases

File: packages/amplify-graphql-model-transformer/src/graphql-model-transformer.ts (hunk 2 above)
Comment (lazpavel) [reply packages/amplify-graphql-model-transformer/src/graphql-model-transformer.ts:371]: also fieldName parameter here is already altered (aka: op + Field / or user custom query name) to match the V1 logic I have no data here

Comment (edwardfoyle) [reply packages/amplify-graphql-model-transformer/src/graphql-model-transformer.ts:371]: ah I see. Okay I guess that makes sense. Just seems like this requires updates in a lot of places and if we could cut that down at all that would be preferable

Comment (lazpavel) [reply packages/amplify-graphql-model-transformer/src/graphql-model-transformer.ts:371]: open for suggestions :) wasn't my first approach, this was the least amount of changes so far

File: packages/amplify-graphql-auth-transformer/src/utils/schema.ts [hunk 3]
```diff
@@ -34,7 +35,7 @@ export const fieldIsList = (fields: ReadonlyArray<FieldDefinitionNode>, fieldNam
 
 export const getModelConfig = (directive: DirectiveNode, typeName: string, isDataStoreEnabled = false): ModelDirectiveConfiguration => {
   const directiveWrapped: DirectiveWrapper = new DirectiveWrapper(directive);
-  const options = directiveWrapped.getArguments<ModelDirectiveConfiguration>({
+  const options = directiveWrapped.getArguments({
```
Comment (SwaySway) [inline packages/amplify-graphql-auth-transformer/src/utils/schema.ts:38]: Is there a reason we need to remove this type?

File: packages/amplify-graphql-transformer-core/src/transform-host.ts (hunk 1 above)
Comment (SwaySway) [reply packages/amplify-graphql-transformer-core/src/transform-host.ts:134]: I agree with @edwardfoyle we should consider it option since there a good chunk of instances where we use the actual typeName and fieldName, in areas where don't actually provide the typeName we should use the resolverLogicalId as an override.

File: packages/amplify-graphql-auth-transformer/src/utils/schema.ts (hunk 3 above)
Comment (lazpavel) [reply packages/amplify-graphql-auth-transformer/src/utils/schema.ts:38]: it doesn't, updated

File: packages/amplify-graphql-auth-transformer/src/utils/schema.ts [hunk 4]
```diff
@@ -234,28 +235,29 @@ export const addDirectivesToOperation = (
...
 export const getQueryFieldNames = (
   modelDirectiveConfig: ModelDirectiveConfiguration,
+  typeName: string,
 ): Set<{ fieldName: string; typeName: string; type: QueryFieldType }> => {
   const fields: Set<{ fieldName: string; typeName: string; type: QueryFieldType }> = new Set();
   if (modelDirectiveConfig?.queries?.get) {
     fields.add({
       typeName: 'Query',
-      fieldName: modelDirectiveConfig.queries.get,
+      fieldName: modelDirectiveConfig.queries.get || toCamelCase(['get', typeName]),
```
[the remark this answers is not in the document]
Comment (lazpavel) [reply packages/amplify-graphql-auth-transformer/src/utils/schema.ts:244]: updated

File: packages/amplify-graphql-transformer-core/src/utils/directive-wrapper.ts [hunk 5]
```diff
@@ -23,7 +22,7 @@ export class DirectiveWrapper {
   private location?: Location;
   constructor(node: DirectiveNode) {
     this.name = node.name;
-    this.arguments = (node.arguments ?? []).map(arg => new ArgumentWrapper(arg));
+    this.arguments = (node.arguments || []).map(arg => new ArgumentWrapper(arg));
```
Comment (cjihrig) [inline packages/amplify-graphql-transformer-core/src/utils/directive-wrapper.ts]: Why is this change necessary? What falsy values can `node.arguments` have?

File: packages/amplify-e2e-tests/src/__tests__/transformer-migrations/model-migration.test.ts [hunk 6]
```diff
@@ -0,0 +1,272 @@
...
+  amplifyPushUpdate,
+  addFeatureFlag,
+  createRandomName,
+  addAuthWithDefault,
+} from 'amplify-e2e-core';
+import { addApiWithoutSchema, updateApiSchema, getProjectMeta } from 'amplify-e2e-core';
+import { createNewProjectDir, deleteProjectDir } from 'amplify-e2e-core';
+import AWSAppSyncClient, { AUTH_TYPE } from 'aws-appsync';
+import gql from 'graphql-tag';
+(global as any).fetch = require('node-fetch');
```
Comment (cjihrig) [inline packages/amplify-e2e-tests/src/__tests__/transformer-migrations/model-migration.test.ts:14]: Is there any way to do this without putting fetch on the global object? It's not the end of the world, just not ideal.

File: packages/amplify-e2e-tests/src/__tests__/transformer-migrations/model-migration.test.ts [hunk 7]
```diff
@@ -0,0 +1,272 @@
...
+      fetchPolicy: 'no-cache',
+    });
+
+    expect(queryResult.errors).toBeUndefined();
+    expect(queryResult.data).toBeDefined();
+  });
+
+  const getAppSyncClientFromProj = (projRoot: string) => {
+    const meta = getProjectMeta(projRoot);
+    const region = meta['providers']['awscloudformation']['Region'] as string;
```
Comment (cjihrig) [inline packages/amplify-e2e-tests/src/__tests__/transformer-migrations/model-migration.test.ts:257]:
```suggestion
    const region = meta.providers.awscloudformation.Region as string;
```

File: packages/amplify-graphql-transformer-core/src/transform-host.ts (hunk 1 above)
Comment (lazpavel) [reply packages/amplify-graphql-transformer-core/src/transform-host.ts]: updated

File: packages/amplify-graphql-transformer-core/src/utils/directive-wrapper.ts (hunk 5 above)
Comment (lazpavel) [reply packages/amplify-graphql-transformer-core/src/utils/directive-wrapper.ts]: just a copy paste difference, reverted

File: packages/amplify-graphql-model-transformer/src/wrappers/object-definition-wrapper.ts [hunk 8]
```diff
@@ -32,50 +29,6 @@ import {
 // Todo: to be moved to core later. context.output.getObject would return wrapper type so its easier to manipulate
 // objects
```
Comment (SwaySway) [inline packages/amplify-graphql-model-transformer/src/wrappers/object-definition-wrapper.ts:30]: nit: leftover comment

**Review by SwaySway — APPROVED** [review]
lgtm aside from nits

File: packages/amplify-e2e-tests/src/__tests__/transformer-migrations/model-migration.test.ts (hunk 6 above)
Comment (lazpavel) [reply packages/amplify-e2e-tests/src/__tests__/transformer-migrations/model-migration.test.ts:14]: didn't spend too much investigating this, it fails without it and we have this line in every test we use the appSyncClient
